lib/ace/mode/rholang_highlight_rules.js

```diff
@@ -70,7 +70,7 @@ var RholangHighlightRules = function() {
             }, {
                 include: "#comment"
             }, {
-                include: "#block-declaration"
+                // include: "#block-declaration"
```

Comment:
Hey Will, good catch for this fix. I gave up after a few hours. :)

Only one little thing. Now `for` inner block is not highlighted correctly because of the end-regex for the whole `for` declaration which is `}`, defined in line 56. If we change it to `/(?<=\{)/` then only matches `for (...bind...)` and not `{...}` block, which will be matched by other rules. I was expecting to be `#block-declaration` but it seems that `{}` are recognized as _text_ scope. Maybe for Ace we need to make a simpler version where `{...}` block is not part of any rule.

![for-inner-block](https://user-images.githubusercontent.com/5306205/37115665-13497d50-224c-11e8-887e-abf85ab6b372.png)

If you have time you can push your fork directly to Ace repo. Just keep in mind to remove `./tool/Rholang.tmLanguage` which I added only for development.

Reply:
yes, you are right. I would dig into the for description. I think we are close to the answer

Reply:
But I think it's good enough for the first version. We can fix these details later. I hope that we can improve tmLanguage version and just generate it for multiple editors without modification so we can easily automate things.